src/include/di/di_help.h

```diff
@@ -0,0 +1,269 @@
+};
+
+/**
+ * This policy ensures that no default values is used, and all parameters being injected are bound
+ */
+// TODO(Tianyu): I believe this will just ensure there is at least a bind clause for anything injected.
+// Does't matter whether it's in() or to().
+class StrictBindingPolicy : public di::config {
+ public:
+  /**
+   * @param ... vararg input
+   * @return strict binding policy
+   */
+  static auto policies(...) noexcept {
+    using namespace di::policies;  // NOLINT
+    return di::make_policies(constructible(is_bound<di::_>{}));
+  }
+};
+
+/**
+ * This policy ensures that all named values is bound. It is okay if some values are default.
+ */
+class TestBindingPolicy : public di::config {
+ public:
+  /**
+   * @param ... vararg input
+   * @return strict binding policy
+   */
+  static auto policies(...) noexcept {
+    using namespace di::policies;             // NOLINT
+    using namespace di::policies::operators;  // NOLINT
+    // Unnamed unbound variables are most likely not
+    return di::make_policies(constructible(is_bound<di::_>{} || !named<di::_>{}));
+  }
+};
+
+/**
+ * Custom wrapper type that allows conversion to *, ManagedPtr and const &, in accordance
+ * with the terrier code base
+ * @tparam TExpected the exposed type to be injected (interface type)
+ * @tparam TGiven the underlying implementation type
+ */
+template <class TExpected, class TGiven>
+class TerrierWrapper {
+ public:
+  /**
+   * @param object underlying object
+   */
+  TerrierWrapper(TExpected *object) : wrapped(object) {}  // NOLINT
+
+  /**
+   * @tparam I target managed pointer's underlying type
+   * @return cast to managed pointer
+   */
+  template <class I, __BOOST_DI_REQUIRES(di::aux::is_convertible<TExpected *, I *>::value) = 0>
+  inline operator common::ManagedPointer<I>() const noexcept {  // NOLINT
+    return common::ManagedPointer<I>(wrapped);
+  }
+
+  /**
+   * @return cast to raw pointer
+   */
+  inline operator TExpected *() const noexcept { return wrapped; }  // NOLINT
+
+  /**
+   * @return cast to constant reference
+   */
+  inline operator const TExpected &() const noexcept {  // NOLINT
+    return *wrapped;
+  }
+
+ private:
+  TExpected *wrapped;
+};
+
+/**
+ * Custom scope for boost::di that corresponds a module in the terrier system.
+ *
+ * This module injects objects with lifetime the same as the injector. All injected object
+ * share the same instance if injected from the same injector. Think of this as a singleton
+ * that has the lifetime of the injector instead of the process.
+ */
+class TerrierSharedModule {
+ public:
+  /**
+   * Implementation of the scope. see boost::di doc
+   * @tparam TExpected
+   * @tparam TGiven
+   */
+  template <class TExpected, class TGiven>
+  class scope {
+   public:
+    // TODO(Tianyu): Not sure about this. This is the referrable flag used for boost::di's singleton scope.
+    /**
+     * See boost::di doc
+     */
```

Comment:
Use `@see` with a link to the specific documentation page.

Reply:
@see only works for code concepts I think? I will just link to their webpage here.